odoo/addons/base/models/res_users.py

```diff
@@ -1654,6 +1654,9 @@ def remove(self):
     def _remove(self):
         """Use the remove() method to remove an API Key. This method implement logic,
         but won't check the identity (mainly used to remove trusted devices)"""
+        if not len(self):
```

Comment:
Just `not self` suffices.

Also not sure there's a point to returning something here? its `remove` doesn't use it, neither does `_revoke_all_devices`. Or is `remove` missing a `return`?

Reply:
Then `remove` should return the action dict too as it does before ?
So the code would be this one ?
```python
@check_identity
def remove(self):
    return self._remove()
```

Reply:
If the intent is to return `close` when clicking on the button then yes, otherwise the return value is just lost. That's only useful if the button invoking `remove` is in a dialog though.